### PR TITLE
Modify the best practice guider for multi-node gpu training

### PR DESCRIPTION
**Pull request by sandyhouse**
Modify the best practice guider for multi-node gpu training.
1. change pyreader to DataLoader
2. remove some old content
3. add the link for dali

式GPU训练优秀实践-使用文档

### CODE REVIEW

File: doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst [hunk 1]
```diff
@@ -7,34 +7,48 @@
...
+数据读取的优化在GPU训练中至关重要，尤其在不断增加batch_size提升吞吐时，计算对reader性能会有更高对要求，
+优化reader性能需要考虑的点包括：
 
-..  csv-table:: 
-    :header: "调节项", "可选值", "说明"
-    :widths: 3, 3, 5
+ - 使用 :code:`DataLoader` 。参考 `这里 <https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api_cn/io_cn/DataLoader_cn.html#dataloader>`_ 使用DataLoader，并建议开启 :code:`use_double_buffer` 。
+ - reader返回uint8类型数据。图片在解码后一般会以uint8类型存储，如果在reader中转换成float类型数据，会将数据体积扩大4倍。直接返回uint8数据，然后在GPU上转化成float类型进行训练可以提升数据读取效率。
+ - 减少reader初始化时间 (infinite read)。
+   在训练任务开始执行第一轮训练时，reader开始异步地不断的从磁盘或其他存储中读取数据并执行预处理，然后将处理好的数据
```
Comment (Heeenrrry) [inline doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst]: 不断异步地

File: doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst [hunk 2]
```diff
@@ -7,34 +7,48 @@
...
+   填充到队列中供计算使用。从0开始填充这个队列直到数据可以源源不断供给计算，需要一定时间的预热。所以，如果每轮训练
```
Comment (Heeenrrry) [inline doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst]: 换行导致空格

File: doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst [hunk 3]
```diff
@@ -7,34 +7,48 @@
...
+   都重新填充队列，会产生一些时间的开销。所以，在使用DataLoader时，可以让reader函数不断的产生数据，直到训练循环结束：
```
Comment (Heeenrrry) [inline doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst]: 换行导致空格；不断地

Comment (Heeenrrry) [reply doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst]: 请更新

File: doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst (hunk 2 above)
Comment (sandyhouse) [reply doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst]: done

File: doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst (hunk 3 above)
Comment (sandyhouse) [reply doc/fluid/advanced_guide/performance_improving/multinode_training_improving/dist_training_gpu.rst]: done

**Review by Heeenrrry — APPROVED** [review]
LGTM

**Pull request merged**